src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -0,0 +1,198 @@
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -p' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_pools,basic
+        """
+        # Create pool and get the storage smd information, then verfify info
+        self.prepare_pool(self.dmg)
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+        self.check_smd_out(smd_info, "pools", "devices")
+
+        # Destroy pool and get the storage smd information, then verify info
+        self.pool.destroy()
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+        self.check_smd_out(smd_info, "devices", "pools")
+
+    def test_dmg_storage_query_blobstore_health(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query blobstore-health' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,blobstore,basic
+        """
+        # Create pool and get the storage smd information
+        self.prepare_pool(self.dmg)
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+
+        # Parse stdout into a dict check that devices information is displayed.
+        blobstore_info = []
+        for info in smd_info:
+            if "Device" in info:
+                uuid = info[info.index("Device") + 1]
+                blobstore_info.append(
+                    self.get_dmg_cmd_info("storage_query_blobstore", uuid))
+
+        # Compare config expected values with dmg output
+        e_blob_info = self.params.get("blobstore_info", "/run/*")
+        if blobstore_info.sort() != e_blob_info.sort():
+            self.fail("dmg storage query expected output: {}".format(
+                e_blob_info))
+
+    def test_dmg_storage_query_device_state(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query device-state' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,device_state,basic
+        """
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+
+        # Check that device is in a normal state
+        self.check_dev_state_out(
+            smd_info, "storage_query_device_state", "NORMAL")
+
+        # Set device to faulty state and check that it's in FAULTY state
+        self.check_dev_state_out(smd_info, "storage_set_faulty", "FAULTY")
+        self.check_dev_state_out(
+            smd_info, "storage_query_device_state", "FAULTY")
+
+    def test_dmg_storage_set_nvme_fault(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query nvme-faulty' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_set,nvme_faulty,basic
+        """
+        # Set nvme-faulty command to run without devuuid provided.
+        self.set_sub_command("storage")
+        self.sub_command_class.set_sub_command("set")
+        self.sub_command_class.sub_command_class.set_sub_command("nvme-faulty")
```

Comment:
(pylint-no-member) Instance of 'DmgStorageQuery' has no 'sub_command_class' member